great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -2485,7 +2524,9 @@ def get_available_data_asset_names(
 
             elif len(batch_kwargs_generator_names) == 1:
                 datasource = self.get_datasource(datasource_names[0])
-                datasource_names[
+                # 20230120 - Chetan - I believe this is a latent bug - we should not be doing string-based indexing
+                #                     within a list. This will result in a runtime error.
+                datasource_names[  # type:ignore[call-overload]
```

Comment:
Bug to be filed to investigate this - leaving it alone as it has been sitting around for some time and requires some domain knowledge to fully understand

Reply:
Thank you for catching this @cdkini